### DISCUSSION TITLE
Trouble with basic maturin build on py3.10 and apple silicon

### DISCUSSION THREAD

**Discussion by lucidguppy** [PyO3/maturin#835] [Q&A] [answered]
Is there a configuration - my version of python is 3.10.2 universal binary on mac os 12.2.1.  

Is there any way to get maturin to target apple silicon?

```
maturin new -b pyo3 guessing_game
maturin develop --cargo-extra-args="--target aarch64-apple-darwin"
```

Output
```
📦 Built wheel for CPython 3.10 to /var/folders/5k/fg4jvbmj1279mc0bg8r314s80000gn/T/.tmps7TdUL/guessing_game-0.1.0-cp310-cp310-macosx_10_7_x86_64.whl
💥 maturin failed
  Caused by: pip install in /Users/foo/work/maturin_prac/venv failed running ["-m", "pip", "--disable-pip-version-check", "install", "--force-reinstall", "/var/folders/5k/fg4jvbmj1279mc0bg8r314s80000gn/T/.tmps7TdUL/guessing_game-0.1.0-cp310-cp310-macosx_10_7_x86_64.whl"]: exit status: 1
--- Stdout:

--- Stderr:
ERROR: guessing_game-0.1.0-cp310-cp310-macosx_10_7_x86_64.whl is not a supported wheel on this platform.
---

```

**Comment by messense** [comment, marked as answer]
If you are targeting Apple Silicon from x86_64 macOS, you should use `maturin build` command instead of `maturin develop`, `maturin develop` is for developing on local machine only.

**Reply by dmort27** [reply]
When I following these steps:
```
maturin new -b pyo3 guessing_game
cd guessing_game
maturin build --cargo-extra-args="--target aarch64-apple-darwin"
```
I get x86_64 wheels only:

* guessing_game-0.1.0-cp38-cp38-macosx_10_7_x86_64.whl
* guessing_game-0.1.0-cp39-cp39-macosx_10_7_x86_64.whl

My Pythons (3.8 and 3.9) are both universal. Do I need to do something else in order to get an aarch64 wheel?

Edit: I'm building on an M1 Mac.

**Reply by messense** [reply]
Use `maturin build --target aarch64-apple-darwin`, no need for `--cargo-extra-args`.

**Reply by dmort27** [reply]
Thanks. That works! 

Any ideas about how to make `maturin develop` work (along the same vein)? When I do:

```
maturin new -b pyo3 guessing_game
cd guessing_game
maturin develop
```
I get
```
Built wheel for CPython 3.8 to /var/folders/kl/8qdy4n55727fq6437z7s7jdh0000gn/T/.tmpGUghp7/guessing_game-0.1.0-cp38-cp38-macosx_10_7_x86_64.whl
maturin failed
  Caused by: pip install in /Users/mortensen/mambaforge/envs/panphon2-3.8 failed running ["-m", "pip", "--disable-pip-version-check", "install", "--force-reinstall", "/var/folders/kl/8qdy4n55727fq6437z7s7jdh0000gn/T/.tmpGUghp7/guessing_game-0.1.0-cp38-cp38-macosx_10_7_x86_64.whl"]: exit status: 1
--- Stdout:

--- Stderr:
ERROR: guessing_game-0.1.0-cp38-cp38-macosx_10_7_x86_64.whl is not a supported wheel on this platform.
---
```
`develop` is building an `x86_64` wheel which pip is refusing to install. But `maturin develop` doesn't accept `--target` as an option, of course.

**Reply by messense** [reply]
Run `rustc -vV` and inspect the `host: ` line, I suspect that you installed x86_64 rustc as default toolchain.

**Reply by dmort27** [reply]
You're quite right, again. That worked perfectly!

**Reply by jblindsay** [reply]
Just to let you know, I've finally got it to work. I just downloaded the latest version of Python from the Python webpage and ensured that it was the 'Universal2' version. I then created a new venv, reinstalled Maturin there, which thankfully was v0.13, and then everything worked just fine. The brew install of Maturin didn't update either, for reasons I can't quite figure out, but this approach did work. Anyhow, I could never have got this fixed without your help and so truly would like to thank you for spending the time with me today.
